src/main/modules/awilix/index.ts

```diff
@@ -27,21 +21,27 @@ const logger: LoggerInstance = Logger.getLogger('app');
  */
 export class Container {
   public enableFor(app: Application): void {
-    app.locals.container = createContainer({ injectionMode: InjectionMode.CLASSIC }).register({
+    const stepControllers = getSteps().map((step: Step) => {
+      const stepDir = `${__dirname}/../../steps/sequence/${step.id}`;
+      const view = `${stepDir}/template.njk`;
+      const { generateContent, form } = require(`${stepDir}/content.ts`);
+
+      return {
+        [`${step.id}StepGetController`]: asValue(
+          new GetController(fs.existsSync(view) ? view : `${stepDir}/../template.njk`, generateContent(step.title))
+        ),
+        ...(form && { [`${step.id}StepPostController`]: asValue(new PostController(new Form(form))) }),
+      };
+    });
+
+    const baseDependencies = {
       logger: asValue(logger),
       homeGetController: asValue(new HomeGetController()),
       termsAndConditionsGetController: asValue(new TermsAndConditionsGetController()),
-      languagePreferenceGetController: asValue(new LanguagePreferenceGetController()),
-      languagePreferencePostController: asValue(new LanguagePreferencePostController(new Form(languagePreferenceForm))),
-      hasMarriageBrokenGetController: asValue(new HasMarriageBrokenGetController()),
-      hasMarriageBrokenPostController: asValue(new HasMarriageBrokenPostController(new Form(hasMarriageBrokenForm))),
-      respondentAddressGetController: asValue(new RespondentAddressGetController()),
-      respondentAddressPostController: asValue(new RespondentAddressPostController(new Form(respondentAddressForm))),
-      marriageCertificateGetController: asValue(new MarriageCertificateGetController()),
-      marriageCertificatePostController: asValue(
-        new MarriageCertificatePostController(new Form(marriageCertificateForm))
-      ),
       errorController: asClass(ErrorController),
-    });
+    };
+    const dependencies = Object.assign(baseDependencies, ...stepControllers);
```

Comment:
rather than mapping the steps above and then injecting them you could just for loop and mutate... but I've already messed you around twice.

Reply:
👍 Fixed in d92f164